server/py/framework/db/sqldb/db.py

```diff
@@ -1515,7 +1515,7 @@ def _find_artifacts(
         query = session.query(ArtifactV2).with_entities(
             ArtifactV2.id,
             ArtifactV2.Tag.name,
-        )
+        ).with_hint(ArtifactV2, "USE INDEX idx_project_bi_updated")
```

Comment:
We don't want to use this index for all queries, since this bypasses the DB optimizer - please see in the discussion in the Jira - 

> the challenge is to correctly tell the DB when to use these indexes. We need to be very careful not to interfere too much with the DB’s optimizer, and intervene only when we know we’re right.

This needs to be added only if doing pagination or ordering by updated. Possibly other conditions.